### ISSUE TITLE
Report if generic methods return objects but perform casts before returning them

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#598]
We should report warnings in case a generic method returns `object` but within the code, casts are done before returning the values.

For example, following code should trigger a warning:
```c#
        private object ReturnSomething<T>(int someValue)
        {
            if (IsAcceptable(typeof(T)))
            {
                return (T) ReturnSomething_A(someValue);
            }

            return (T) ReturnSomething_B(someValue);
        }
```
There is no need in the cast when the code itself then casts back to object.
